tern/formats/spdx_new/package_helpers.py

```diff
@@ -0,0 +1,168 @@
+        copyright_text=mapping['PackageCopyrightText'] if mapping['PackageCopyrightText'] else SpdxNone(),
+        comment=SOURCE_PACKAGE_COMMENT,
+    )
+
+
+def get_package_dict(package: Package, template: Template) -> SpdxPackage:
+    """Given a package object and its SPDX template mapping, return an SPDX Package.
+    The analyzed files will go in a separate dictionary for the JSON document."""
+    mapping = package.to_dict(template)
+
+    if mapping['PackageSupplier']:
+        supplier = Actor(ActorType.ORGANIZATION, mapping['PackageSupplier'])
+    else:
+        supplier = SpdxNoAssertion()
+
+    external_ref = []
+    if get_purl(package):
+        external_ref.append(ExternalPackageRef(
+            ExternalPackageRefCategory.PACKAGE_MANAGER,
+            "purl",
+            get_purl(package)
+        ))
+
+    pkg_ref, _ = get_package_spdxref(package)
+    # Define debian licenses from copyright text as one license
+    declared_lic = mapping['PackageLicenseDeclared']
+    if package.pkg_format == 'deb':
+        declared_lic = ', '.join(package.pkg_licenses)
+
+    return SpdxPackage(
+        spdx_id=pkg_ref,
+        name=mapping['PackageName'],
+        version=mapping['PackageVersion'] if mapping['PackageVersion'] else 'NOASSERTION',
+        supplier=supplier,
+        download_location=mapping['PackageDownloadLocation'] if mapping['PackageDownloadLocation'] else SpdxNoAssertion(),
+        files_analyzed=False,
+        license_concluded=SpdxNoAssertion(),
+        license_declared=get_package_license_declared(declared_lic),
+        copyright_text=mapping['PackageCopyrightText'] if mapping['PackageCopyrightText'] else SpdxNone(),
+        external_references=external_ref,
+        comment=get_package_comment(package),
+    )
+
+
+def get_packages_list(image_obj: Image, template: Template) -> List[SpdxPackage]:
+    """Given an image object and the template object for SPDX, return a list
+    of SPDX dictionary representations for each of the packages in the image.
+    The SPDX JSON spec for packages requires:
+        name
+        versionInfo
+        downloadLocation"""
+    packages = []
+    package_refs = set()
+
+    for layer in image_obj.layers:
+        for package in layer.packages:
+            # Create a list of dictionaries. Each dictionary represents
+            # one package object in the image
+            pkg_ref, src_ref = get_package_spdxref(package)
+            if pkg_ref not in package_refs and package.name:
+                packages.append(get_package_dict(package, template))
+                package_refs.add(pkg_ref)
+            if src_ref and src_ref not in package_refs:
+                packages.append(get_source_package_dict(
+                    package, template))
+                package_refs.add(src_ref)
+    return packages
+
+
+purl_types_with_namespaces = [
+    'deb',
+    'rpm',
+    'apk',
+    'alpm'
+]
+
+
+def get_purl(package_obj: Package) -> str:
```

Comment:
This function needs updating per: https://github.com/tern-tools/tern/pull/1237/files#diff-8853c35f8baed4fa963a218102cecf6fc9153733acaaf79b6210af00bc44e068

Reply:
updated